src/main/java/persistence/entity/EntityMeta.java

```diff
@@ -0,0 +1,51 @@
+package persistence.entity;
+
+import persistence.persistencecontext.EntityKey;
+import persistence.sql.domain.IdColumn;
+import persistence.sql.domain.Table;
+import utils.ValueExtractor;
+import utils.ValueInjector;
+
+public class EntityMeta {
+    private final Table table;
+    private final IdColumn id;
+    private final Object entity;
+    private final EntityEntry entityEntry;
+
+    public EntityMeta(Table table, IdColumn id, Object entity) {
+        this.table = table;
+        this.id = id;
+        this.entity = entity;
+        this.entityEntry = new EntityEntry();
+    }
+
+    public static <T> EntityMeta from(T entity) {
+        Table table = Table.from(entity.getClass());
+        IdColumn id = table.getIdColumn();
+        return new EntityMeta(table, id, entity);
+    }
+
+    public EntityKey getEntityKey() {
+        return new EntityKey(getId(), entity.getClass());
+    }
+
+    public boolean isNew() {
+        return getId() == null;
+    }
+
+    public Object getId() {
+        return ValueExtractor.extract(entity, id);
+    }
```

Comment:
```suggestion
    public boolean isNew(Object entity) {
        return ValueExtractor.extract(entity, id) == null;
    }
```
entity 는 엔터티 메타정보가 아니다 보니 메시지로 전달하여 처리하도록 변경하는게 더 객체의 책임에 맞다고 느껴지는데
현태님은 혹시 어떻게 생각하시나요?

Reply:
EntityMeta에 맞도록 수정하도록 하겠습니다!